level.py

```diff
@@ -0,0 +1,94 @@
+import pygame
+from constants import WIDTH, HEIGHT, block_size, BLACK
+
+LEVELS = [
+
+    ["------------------------------",
+     "------------------------------",
+     "------------------------------",
+     "----------1------1------------",
+     "----------1------1------------",
+     "----------1------1------------",
+     "----------1------1------------",
+     "------11111------11111--------",
+     "------------------------------",
+     "------------------------------",
+     "------------------------------",
+     "------------------------------",
+     "------11111------11111--------",
+     "----------1------1------------",
+     "----------1------1------------",
+     "----------1------1------------",
+     "----------1------1------------",
+     "------------------------------",
+     "------------------------------",
+     "------------------------------"
+     ],
+
+     ["------------1111--------------",
+     "------------------------------",
+     "------------------------------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "-----1----1------1----1-------",
+     "------------------------------",
+     "------------------------------",
+     "------------1111--------------"
+     ],
+
+     ["------------------------------",
+     "---11111----------------------",
+     "---1---1----------------------",
+     "------------------------------",
+     "---1---1-----1--1-------------",
+     "---11111-----1--1-------------",
+     "-------------1--1-------------",
+     "-------------1--1-------------",
+     "---------11111--11111---------",
+     "------------------------------",
+     "------------------------------",
+     "---------11111--11111---------",
+     "-------------1--1-------------",
+     "-------------1--1-------------",
+     "-------------1--1-----11111---",
+     "-------------1--1-----1---1---",
+     "------------------------------",
+     "----------------------1---1---",
+     "----------------------11111---",
+     "------------------------------"
+     ]
+]
+
+class Level:
+    def __init__(self):
+        self.level_index = 0
+        self.map = LEVELS[self.level_index]
+        self.level_goal = (self.level_index + 1) * 50
+
+    def draw_level(self, screen):
+        for y in range(HEIGHT // block_size):
+            for x in range(WIDTH // block_size):
+                if self.map[y][x] in "123":
```

Comment:
I guess you're just using `1` right?

Reply:
A simpler idea is to simply respawn any neggs that happen to get spawned in a wall until they land outside a wall. You could just use the existing Level.is_wall function to know if a negg needs to try again.